smartling-api-commons/src/test/java/com/smartling/api/v2/auth/AuthenticationIntegrationTest.java

```diff
@@ -0,0 +1,203 @@
+    {
+        @Override
+        protected Class<DummyApi> getApiClass()
+        {
+            return DummyApi.class;
+        }
+    };
+
+    @Before
+    public void setup()
+    {
+        smartlingApi.stubFor(get(urlEqualTo(DUMMY_API)).willReturn(ok()));
+    }
+
+    private DummyApi dummyApi() throws Exception
+    {
+        return dummyApi("user", "secret");
+    }
+
+    private DummyApi dummyApi(String user, String secret) throws Exception
+    {
+        ClientFactory clientFactory = new ClientFactory();
+        ClientConfiguration configuration = DefaultClientConfiguration.builder()
+            .baseUrl(new URL(smartlingApi.baseUrl()))
+            .build();
+        AuthenticationApi authenticationApi = new AuthenticationApiFactory(clientFactory)
+            .buildApi(configuration);
+        Authenticator authenticator = new Authenticator(user, secret, authenticationApi);
+
+        return dummyFactory.buildApi(new BearerAuthSecretFilter(authenticator), configuration);
+    }
+
+    @Test
+    public void shouldAuthenticate() throws Exception
+    {
+        smartlingApi.stubFor(post(urlEqualTo("/auth-api/v2/authenticate"))
+            .withHeader("Content-Type", equalTo("application/json"))
+            .withRequestBody(equalToJson("{\n" +
+                "  \"userIdentifier\": \"user\",\n" +
+                "  \"userSecret\": \"secret\"\n" +
+                "}")
+            )
+            // language=JSON
+            .willReturn(okJson("{\n" +
+                "  \"response\": {\n" +
+                "    \"code\": \"SUCCESS\",\n" +
+                "    \"data\": {\n" +
+                "      \"accessToken\": \"accessTokenValue\",\n" +
+                "      \"refreshToken\": \"refreshTokenValue\",\n" +
+                "      \"expiresIn\": 480,\n" +
+                "      \"refreshExpiresIn\": 21600,\n" +
+                "      \"tokenType\": \"Bearer\"\n" +
+                "    }\n" +
+                "  }\n" +
+                "}")
+            )
+        );
+
+        dummyApi().dummy();
+
+        smartlingApi.verify(getRequestedFor(urlEqualTo(DUMMY_API))
+            .withHeader("Authorization", equalTo("Bearer accessTokenValue")));
+    }
+
+    @Test
+    public void shouldThrowValidationErrorExceptionOnInvalidCredentials() throws Exception
+    {
+        smartlingApi.stubFor(post(urlEqualTo("/auth-api/v2/authenticate"))
+            .withHeader("Content-Type", equalTo("application/json"))
+            .withRequestBody(equalToJson("{\n" +
+                "  \"userIdentifier\": \"invalidUser\",\n" +
+                "  \"userSecret\": \"invalidSecret\"\n" +
+                "}")
+            )
+            .willReturn(unauthorized()
+                .withHeader("Content-Type", "application/json")
+                // language=JSON
+                .withBody("{\n" +
+                    "  \"response\": {\n" +
+                    "    \"code\": \"VALIDATION_ERROR\",\n" +
```

Comment:
@foo4u 
This is the real response from auth-api on invalid credentials
Is it possible to get [AUTHENTICATION_ERROR](https://github.com/Smartling/java-api-sdk/blob/f00b29d17c60157375df4f06c629f020d345e0f3/smartling-api-commons/src/main/java/com/smartling/api/v2/response/ResponseCode.java#L8) here from auth-api?

Reply:
It the user provided incorrect credentials, isn't a validation error?

Reply:
Changing error types is potentially an API breaking change for consumers due to a change in the contract of the response.  I would be very careful with this type of suggestion.

Reply:
I'm with Ben on not changing this for 2 reasons:

1. It breaks the current API contract (most importantly)
2. It's not what we intended `AUTHENTICATION_ERROR` for. Providing bad input to any of our APIs, even the auth API, is a `VALIDATION_ERROR`.

Reply:
I agree to not changing it